styx-common/src/main/java/com/spotify/styx/util/BasicWorkflowValidator.java

```diff
@@ -144,11 +137,97 @@ public BasicWorkflowValidator(DockerImageValidator dockerImageValidator) {
       }
     });
 
+    validateDockerExecConf(e, cfg);
+    validateFlyteExecConf(e, cfg);
+    validateConflictingExecConf(e, cfg);
+
     return e;
   }
 
   private static boolean validateServiceAccount(String serviceAccount) {
     var matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(serviceAccount);
     return matcher.matches();
   }
+
+  private void validateDockerExecConf(List<String> errors, WorkflowConfiguration config) {
+    validateDockerParam(errors, "dockerImage", config.dockerImage(),
+        config.dockerExecConf().flatMap(DockerExecConf::dockerImage),
```

Comment:
Do we expect that both of these dockerImages should be present and the same?

Reply:
Yes it is confusing. I will change it to be a clear cut old style or clear cut new style in flyte or docker